AdyenActions/Components/Await/PollingHandlerProvider.swift

```diff
@@ -31,14 +31,14 @@ internal protocol AnyPollingHandlerProvider {
 internal struct PollingHandlerProvider: AnyPollingHandlerProvider {
 
     /// :nodoc:
-    private let environment: APIEnvironment
+    private let apiContext: APIContext
 
     /// :nodoc:
     private let apiClient: AnyRetryAPIClient?
 
     /// :nodoc:
-    internal init(environment: APIEnvironment, apiClient: AnyRetryAPIClient?) {
-        self.environment = environment
+    internal init(apiContext: APIContext, apiClient: AnyRetryAPIClient?) {
```

Comment:
What is the purpose of this struct? It does only one thing - always creates `PollingComponent` 😄

Reply:
It is an implementation of `AnyPollingHandlerProvider`. We need this protocol for testing. 
We are passing `nil` for `apiClient` in `QRCodeComponent.swift #77`

Reply:
> We are passing nil for apiClient in QRCodeComponent.swift #77

this is my point - we always passing `nil`..

It could be easier to mock `AnyRetryAPIClient` and pass it is `AnyPollingHandler` rather than mock optional `AnyPollingHandlerProvider` ? 🤔

Reply:
maybe we could do this in another PR